btree/btree.c

```diff
@@ -2233,7 +2233,8 @@ static void generic_move(struct nd *src, struct nd *tgt,
 	while (true) {
 		if (nr == 0 || (nr == NR_EVEN &&
 			       (node_space(tgt) <= node_space(src))) ||
-			       (nr == NR_MAX && srcidx == -1))
+			       (nr == NR_MAX && (srcidx == -1 ||
+			       node_count_rec(src) == 0)))
```

Comment:
node_count_rec(src) == 0 is needed as If dir = D_RIGHT , and nr=NR_MAX and  if we move all records, so src will contain zero records and we need come out of the loop. Earlier, conditions were not satisfying this case